addon/components/truncate-multiline.js

```diff
@@ -193,6 +194,38 @@ export default Ember.Component.extend(ResizeHandlerMixin, {
    */
   resizeOnInsert: false,
 
+  /**
+   * Helper function to determine if an attribute has changed.
+   * @private
+   * @param attrs {Object}
+   * @param name {String}
+   * @return {Boolean}
+   */
+  _didAttrChange(attrs, name) {
+    let oldAttr = attrs.oldAttrs[name];
+    let newAttr = attrs.newAttrs[name];
+
+    if (typeof oldAttr === "undefined" || typeof newAttr === "undefined") {
+      return false;
+    }
+
+    /**
+    * In integration tests, the attributes passed to 'didUpdateAttrs' are objects,
+    * while in the app they are the values of the attributes themselves.
+    *
+    * However, if the value of the passed attribute hasn't changed it is NOT an object,
+    * it is the value itself.
+    */
+    let isTesting = Ember.testing;
+
+    if (isTesting) {
+      oldAttr = oldAttr instanceof Object ? attrs.oldAttrs[name].value : oldAttr;
+      newAttr = newAttr instanceof Object ? attrs.newAttrs[name].value : newAttr;
+    }
```

Comment:
Let's not limit this to testing. Also, let's use short circuiting assignment instead of ternaries in case an attribute is an object, but doesn't have a value property.